pkgs/by-name/wo/wordlists/package.nix

```diff
@@ -0,0 +1,65 @@
+{ lib
+, nmap
+, rockyou
+, runtimeShell
+, seclists
+, symlinkJoin
+, tree
+, wfuzz
+, lists ? [
+    nmap
+    rockyou
+    seclists
+    wfuzz
+  ]
+}:
+
+symlinkJoin rec {
+  pname = "wordlists";
+  version = "unstable-2023-10-10";
+
+  name = "${pname}-${version}";
+  paths = lists;
+
+  postBuild = ''
+    # Create a command to show the location of the links.
+    mkdir -p $out/bin
+    cat >> $out/bin/wordlists << __EOF__
+    #!${runtimeShell}
+    ${tree}/bin/tree $out/share/wordlists
+    __EOF__
+    chmod +x $out/bin/wordlists
+
+    # Create a handy command for easy access to the wordlists.
+    # e.g.: `cat "$(wordlists_path)/rockyou.txt"`, or `ls "$(wordlists_path)/dirbuster"`
+    cat >> $out/bin/wordlists_path << __EOF__
+    #!${runtimeShell}
+    echo $out/share/wordlists
+    __EOF__
+    chmod +x $out/bin/wordlists_path
+  '';
```

Comment:
that sounds like a gerate idea, we can just add two extra files containing the derivations with writeShellScript and just do `callPackage` in the package.nix. There is some similar things in nixpkgs already like `pkgs/by-name/wi/windowmaker/dockapps/default.nix`

Reply:
I tried this earlier but it doesn't work since the $out path it then tries to use is the out of it's own derivation and not the out of the symlinkJoin derivation, there might be a way to override that but I don't know of any, so I would leave it in postBuild for now.

Reply:
I've had another look at this. Introducing binaries in the middle of a `symlinkJoin` feels like a hack that circumvents the purpose of `symlinkJoin`.

I think the canonical way would've been to let the original binaries be standalone tools that takes their input via normal methods like args, envvars or stdin, and then rather wrap the program to fix its inputs. Or at least somehow build the tools in standalone derivations that depend on the lists as input.

How about a design that looks more like this?

```mermaid
flowchart TD
    wordlists["wordlists package (symlinkJoin)"]
    binaries["binaries (wrapping wordlist collection dir)"]
    symlinkedWordlists["wordlist collection (symlinkJoin)"]
    wordlist1["wordlist 1"]
    wordlist2["wordlist 2"]
    wordlist3["wordlist 3"]

    wordlists --> binaries
    wordlists --> symlinkedWordlists
    binaries --> symlinkedWordlists
    symlinkedWordlists --> wordlist1
    symlinkedWordlists --> wordlist2
    symlinkedWordlists --> wordlist3
```

Reply:
I get where you coming from with putting binary outputs in to symlinkJoin postBuild being hacky. But you are loosing at the design suggestion you made. 
So `wordlists package (symlinkJoin)` would be a package including wordlists or would that be the current wordlists package we have? 
`binaries (wrapping wordlist collection dir)` this would be binaries interacting with wordlists like the current `wordlists` and `wordlists_path` ones? 
What is `wordlist collection (symlinkJoin)` supposed to be?

Reply:
Let's move over to the other thread